src/event_sources/ble_esp32.cc

```diff
@@ -235,6 +235,40 @@ void BLEEventSource::on_gatt_attribute_event(uint16_t conn_handle,
   dispatch(locker, gatt, 0);
 }
 
+int BLEEventSource::on_gatt_server_characteristic(uint16_t conn_handle, uint16_t attr_handle,
+                                                  struct ble_gatt_access_ctxt* ctxt, void* arg) {
+  auto characteristic = reinterpret_cast<BLEServerCharacteristicResource*>(arg);
+  return instance()->on_gatt_server_characteristic_event(ctxt, characteristic);
+}
+
+int BLEEventSource::on_gatt_server_characteristic_event(ble_gatt_access_ctxt* ctxt,
+                                                        BLEServerCharacteristicResource* characteristic) {
+  int err = 0;
+  switch (ctxt->op) {
+    case  BLE_GATT_ACCESS_OP_READ_CHR:
+      if (characteristic->mbuf_to_send() != null) {
+        err = os_mbuf_appendfrom(ctxt->om, characteristic->mbuf_to_send(), 0, characteristic->mbuf_to_send()->om_len);
+      }
+      break;
+    case BLE_GATT_ACCESS_OP_WRITE_CHR:
+      characteristic->set_mbuf_received(ctxt->om);
+      ctxt->om = null;
+      break;
+    case BLE_GATT_ACCESS_OP_READ_DSC:
+    case BLE_GATT_ACCESS_OP_WRITE_DSC:
+      // ignore
+      return 0;
+    default:
+      // unknown
```

Comment:
Improve comment. What do we do with unknown stuff? It looks like we go on and dispatch, without taking any further action here.